### PR TITLE
Rename the “dependents” section as ”linked_definitions”

### PR DESCRIPTION
**Pull request by paulhuggett**
The change is intended to avoid confusion between the use of the term “dependents” in the context of repository hash generation and the storage of definitions created as a result of global optimization here. It matches the code to the terminology documented on [this wiki page](https://github.com/SNSystems/llvm-project-prepo/wiki/Dealing-with-the-Effects-of-Global-Optimisations).

A couple of things to note:

- This is the first of (probably) two PRs which are intended to enable completion of the work in the [exchange branch](https://github.com/SNSystems/pstore/tree/exchange). The commits there implement import and export to resolve issue  #15 and to make it possible to write LIT tests for the llvm-project-prepo repo2obj and rld tools. The second will be in the same vein as [PR#63](https://github.com/SNSystems/pstore/pull/63) to add information beyond a raw pointer to each record of this structure.

- This PR is paired with an [equivalent PR](https://github.com/SNSystems/llvm-project-prepo/pull/120) in llvm-project-prepo. This PR changes the pstore API: that related PR updates LLVM to use the newer names.

### CODE REVIEW

File: include/pstore/mcrepo/linked_definitions_section.hpp [hunk 1]
```diff
@@ -233,16 +238,16 @@ namespace pstore {
...
-        struct section_to_dispatcher<dependents> {
+        struct section_to_dispatcher<linked_definitions> {
             using type = dependents_dispatcher;
         };
 
     } // end namespace repo
 } // end namespace pstore
 
-#endif // PSTORE_MCREPO_DEPENDENTS_SECTION_HPP
+#endif // PSTORE_MCREPO_LINKED_DEFINITIONS_SECTION_HPP
```
Comment (MaggieYingYi) [inline include/pstore/mcrepo/linked_definitions_section.hpp:253]: The linked_definitions class comments might need to be updated as well. For example, line 128, 132, 136, 138 .. in this file (linked_definitions_section.hpp).

**Review by MaggieYingYi — COMMENTED** [review]
Two things might need to be updated as well.
1) should change `dependents_dispatcher` to `linked_definitions_dispatcher`?
2) In the test_fragment.cpp file Should change the test:
`TEST_F (FragmentTest, MakeTextSectionWithDependents)` to:
`TEST_F (FragmentTest, MakeTextSectionWithLinkedDefinitions)`?

File: include/pstore/mcrepo/linked_definitions_section.hpp (hunk 1 above)
Comment (paulhuggett) [reply include/pstore/mcrepo/linked_definitions_section.hpp:252]: Yes, well spotted. I've updated the change with more renaming.

**Review by MaggieYingYi — APPROVED** [review]
LGTM!

**Pull request merged**